frontend/app/routes/$lang+/_protected+/personal-information+/mailing-address+/edit.tsx

```diff
@@ -144,7 +145,34 @@ export async function action({ context: { session }, params, request }: ActionFu
 
   instrumentationService.countHttpStatus('mailing-address.edit', 302);
 
-  // TODO: Save form data with with future implementation updatePersonalInformation()
+  const userInfoToken: UserinfoToken = session.get('userInfoToken');
+  const personalInformationServie = getPersonalInformationService();
+  const personalInformationRouteHelpers = getPersonalInformationRouteHelpers();
+  const personalInformation = await personalInformationRouteHelpers.getPersonalInformation(userInfoToken, params, request, session);
+
+  const newPersonalInformation = { ...personalInformation };
+  newPersonalInformation.mailingAddress = {
+    streetName: parsedDataResult.data.streetName,
+    secondAddressLine: parsedDataResult.data.secondAddressLine,
+    countryId: parsedDataResult.data.countryId,
+    provinceTerritoryStateId: parsedDataResult.data.provinceTerritoryStateId,
+    cityName: parsedDataResult.data.cityName,
+    postalCode: parsedDataResult.data.postalCode,
+  };
+  newPersonalInformation.homeAndMailingAddressTheSame = false;
+  if (parsedDataResult.data.homeAndMailingAddressTheSame) {
+    newPersonalInformation.homeAndMailingAddressTheSame = true;
+    newPersonalInformation.homeAddress = {
+      streetName: parsedDataResult.data.streetName,
+      secondAddressLine: parsedDataResult.data.secondAddressLine,
+      countryId: parsedDataResult.data.countryId,
+      provinceTerritoryStateId: parsedDataResult.data.provinceTerritoryStateId,
+      cityName: parsedDataResult.data.cityName,
+      postalCode: parsedDataResult.data.postalCode,
+    };
+  }
```

Comment:
**Nitpick:** Might be a tad more readable if we destructure the properties from `parsedDataResult.data` and use them.

```typescript
  const { streetName, secondAddressLine, countryId, provinceTerritoryStateId, cityName, postalCode, homeAndMailingAddressTheSame } = parsedDataResult.data;

  const newPersonalInformation = {
    ...personalInformation,
    homeAddress: {
      streetName,
      secondAddressLine,
      countryId,
      provinceTerritoryStateId,
      cityName,
      postalCode,
    },
    homeAndMailingAddressTheSame,
    mailingAddress: homeAndMailingAddressTheSame ? { streetName, secondAddressLine, countryId, provinceTerritoryStateId, cityName, postalCode } : personalInformation.mailingAddress,
  };
```

Same comment can be applied to the `home-address/edit` route.